creature.cpp

```diff
@@ -1144,6 +1144,8 @@ static string getAttackParam(AttackType type) {
     case AttackType::SPELL: return "spell";
     case AttackType::POSSESS: return "touch";
   }
+  assert("__FUNCTION__ case unknown attack type");
+  return "";
```

Comment:
That's the whole point, if there no defaults then the compiler will warn if any enum value is omitted in the switch. The assert will never be triggered unless the value is outside the enum range, usually when a variable was uninitialized or there was memory corruption. I don't think checking for these is as important, and if it is, a simple 'FAIL << variable' without the return statement will do.

As for reading from text files, there value is already checked there, see EnumInfo<T>::fromString in util.h line 964